### ISSUE TITLE
Angular repository not installed in Bazel workspace file

### ISSUE THREAD

**Issue by remkoboschker** [angular/angular#28636]
# 🐞 bug report

### Affected Package
@angular/bazel

### Is this a regression?

no

### Description

I followed the instructions on https://github.com/angular/angular/tree/master/packages/bazel/src/schematics and get the error ```The angular repository is not installed in your Bazel WORKSPACE file```

## 🔬 Minimal Reproduction
```
ng new demo --collection=@angular/bazel --defaults
cd demo
yarn
ng build
ng serve
```
## 🔥 Exception or Error
The angular repository is not installed in your Bazel WORKSPACE file
## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 7.3.1
Node: 10.15.1
OS: darwin x64
Angular: 7.2.4
... animations, bazel, common, compiler, compiler-cli, core
... forms, language-service, platform-browser
... platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.13.1
@angular-devkit/build-angular     0.13.1
@angular-devkit/build-optimizer   0.13.1
@angular-devkit/build-webpack     0.13.1
@angular-devkit/core              7.3.1
@angular-devkit/schematics        7.3.1
@angular/cli                      7.3.1
@ngtools/webpack                  7.3.1
@schematics/angular               7.3.1
@schematics/update                0.13.1
rxjs                              6.3.3
typescript                        3.2.4
webpack                           4.29.0

</code></pre>

**Comment by remkoboschker** [comment]
ping @mgechev or @kyliau

**Comment by dgarzon** [comment]
@remkoboschker I am also running into the same issue, and here are my findings:

I started with a fresh `ng new` application updating the angular version to `7.2.4` in the `WORKSPACE` and changing some things to align with: https://github.com/angular/angular/blob/master/integration/bazel/WORKSPACE, which lead to the same error you are getting.

Then I looked at the git history, and realized https://github.com/angular/angular/commit/0d1e065a1ced6b6dd4f6a100bfca10ec3a2f2928, did not make it to the `7.2.4` release. That commit deprecates the use of `rules_angular_dependencies` in favor of `ng_setup_workspace`, and removes the `rules_typescript` version check among other things, allowing for better compatibility between angular and typescript rules.

In short, version `7.2.4` of `@angular` and `@angular/bazel` are not compatible with newer versions of `rules_nodejs` or `rules_typescript`.

**Proposed Solution**:
* Remove this check as it is not valid anymore in latest `master` so that it get's released in `7.2.5`: https://github.com/angular/angular/blob/master/packages/bazel/check_version.js#L53. This would effectively align `@angular/bazel` with the changes made in the aforementioned commit.

**Comment by mgechev** [comment]
@remkoboschker thanks for pointing this out! @dgarzon thank you for sharing a sample solution.

I'll look at this today and follow up in a few hours.

**Comment by mgechev** [comment]
There were a few broken things that we'll fix in future releases. You can try a working version in [this](https://github.com/mgechev/bazel-app) repo.

You can find the changes I did [here](https://github.com/mgechev/bazel-app/commit/9727db055b18c0e0317dbda1e8284c61066f3501).

**Comment by kyliau** [comment]
Please try the most recent release of `@angular/bazel`, which is 8.0.0-beta.3
If you run into the same error about `angular repository not installed`, please add the following data dependency to the `yarn_install` rule.
```
yarn_install(
    name = "npm",
    package_json = "//:package.json",
    yarn_lock = "//:yarn.lock",
    data = ["@angular//:LICENSE"],
)
```
(This is just so the angular repository is installed prior to yarn_install)
There was a major refactoring in the commit pointed out by @dgarzon, and some bugs slipped through the tests. We'll fix these issues very soon with the next release. Thank you for trying out Bazel!

**Comment by dgarzon** [comment]
@mgechev @kyliau I can confirm the proposed solution by @kyliau got me past the issue with some tweaks to a couple `BUILD.bazel` files. Unfortunately, it introduces another issue when compiling TypeScript:

```
SUBCOMMAND: # @angular//packages/bazel/src/ngc-wrapped:ngc_lib [action 'Compiling TypeScript (devmode) @angular//packages/bazel/src/ngc-wrapped:ngc_lib [for host]']

ERROR: /private/var/tmp/_bazel_danielgarzon/7af0eab87c556eada10f24ad9bc85f00/external/angular/packages/bazel/src/ngc-wrapped/BUILD.bazel:4:1: Compiling TypeScript (devmode) @angular//packages/bazel/src/ngc-wrapped:ngc_lib failed (Exit 1)
external/angular/packages/bazel/src/ngc-wrapped/index.ts:46:9 - error TS2488: Type '[ParsedTsConfig, Diagnostic[], { target: string; }]' must have a '[Symbol.iterator]()' method that returns an iterator.

46   const [parsedOptions, errors] = parseTsconfig(project);
external/angular/packages/bazel/src/ngc-wrapped/index.ts:141:35 - error TS2488: Type 'string[]' must have a '[Symbol.iterator]()' method that returns an iterator.

141   const writtenExpectedOuts = [...expectedOuts];
                                      ~~~~~~~~~~~~
external/angular/packages/bazel/src/ngc-wrapped/index.ts:162:13 - error TS2488: Type 'RegExpExecArray' must have a '[Symbol.iterator]()' method that returns an iterator.

162   const [, file, suffix, ext] = match;
                ~~~~~~~~~~~~~~~~~~~~~
external/angular/packages/bazel/src/ngc-wrapped/index.ts:383:23 - error TS2488: Type 'ReadonlyArray<Diagnostic>' must have a '[Symbol.iterator]()' method that returns an iterator.

383   diagnostics.push(...tsProgram.getOptionsDiagnostics());
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/angular/packages/bazel/src/ngc-wrapped/index.ts:384:23 - error TS2488: Type 'ReadonlyArray<Diagnostic>' must have a '[Symbol.iterator]()' method that returns an iterator.

384   diagnostics.push(...tsProgram.getGlobalDiagnostics());
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/angular/packages/bazel/src/ngc-wrapped/index.ts:390:25 - error TS2488: Type 'ReadonlyArray<DiagnosticWithLocation>' must have a '[Symbol.iterator]()' method that returns an iterator.

390     diagnostics.push(...tsProgram.getSyntacticDiagnostics(sf));
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/angular/packages/bazel/src/ngc-wrapped/index.ts:391:25 - error TS2488: Type 'ReadonlyArray<Diagnostic>' must have a '[Symbol.iterator]()' method that returns an iterator.

391     diagnostics.push(...tsProgram.getSemanticDiagnostics(sf));
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/angular/packages/bazel/src/ngc-wrapped/index.ts:396:25 - error TS2488: Type 'ReadonlyArray<Diagnostic>' must have a '[Symbol.iterator]()' method that returns an iterator.

396     diagnostics.push(...ngProgram.getNgStructuralDiagnostics());
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/angular/packages/bazel/src/ngc-wrapped/index.ts:397:25 - error TS2488: Type 'ReadonlyArray<Diagnostic | Diagnostic>' must have a '[Symbol.iterator]()' method that returns an iterator.

397     diagnostics.push(...ngProgram.getNgSemanticDiagnostics());
```

Any ideas? Thanks in advanced!

**Comment by dgarzon** [comment]
Hi @kyliau, thanks for your suggestion! Now I am able to `build`, `test` and `run` everything except the `devserver`. My `BUILD.bazel` has this entry: 
```
ts_devserver(
    name = "devserver",
    port = 4200,
    additional_root_paths = [
        "npm/node_modules/zone.js/dist",
        "npm/node_modules/tslib",
    ],
    entry_module = "frontend/src/main.dev",
    serving_path = "/bundle.min.js",
    static_files = [
        "@npm//node_modules/zone.js:dist/zone.min.js",
        "@npm//node_modules/tslib:tslib.js",
    ],
    index_html = "index.html",
    deps = [":src"],
)
```
From the Developer Tools console, I see the `devserver` loads `bundle.min.js` and `main.dev.js`, but it is not serving the correct `index.html`, let alone any of the `static_files`. Do you have any idea why this would happen?

**Comment by kyliau** [comment]
We are aware of the issue with `ts_devserver`. @devversion could you please provide an update here?

**Comment by devversion** [comment]
@kyliau Yes. https://github.com/bazelbuild/rules_typescript/pull/385 fixed that issue.
